.github/workflows/preview_cd.yml

```diff
@@ -22,6 +22,13 @@ jobs:
         shell: bash
         run: |
            echo "sha_short=`echo ${{ github.event.pull_request.head.sha }} | cut -c1-8`" >> $GITHUB_ENV
+      - name: Encode URL Branch
+        id: encode_url_branch
+        uses: actions/github-script@v7
+        with:
+          result-encoding: string
+          script: |
+            return "${{ github.head_ref }}".replace(/[^0-9a-zA-Z\-\.]+/g, "-")
```

Comment:
What if `head_ref` contains `"`?

Reply:
I mean at that point, we need to have a talk about naming branches :sob:

Everything should follow the convention defined in `README` and this is really only so that dependabot can work. Even if I do the backtick literal for strings, the branch name could always have a backtick so I feel like this one's a bit unwinnable without complicating it further than it has to be.

Reply:
Yeah... Let's not worry too much about it now. Just that it's potentially hackable.